processor/src/main/java/org/mapstruct/ap/internal/model/source/ForgedMethod.java

```diff
@@ -216,4 +216,9 @@ public MapperConfiguration getMapperConfiguration() {
     public boolean isUpdateMethod() {
         return getMappingTargetParameter() != null;
     }
+
+    @Override
+   	public boolean isObjectFactory() {
+   		return getTargetTypeParameter() != null || getParameters().isEmpty();
```

Comment:
just return `false` here. We wouldn't create a forged factory method, at least not yet.